### PR TITLE
Delete button on Team Card

### PR DESCRIPTION
**Pull request by chengxi600**
## Describe your changes

Added delete button to Teams Card. Moved TeamDelModal to @genshin-optimizer/gi/ui 

## Issue or discord link

Resolves [https://github.com/frzyc/genshin-optimizer/issues/2132](https://github.com/frzyc/genshin-optimizer/issues/2132)

## Testing/validation

https://github.com/frzyc/genshin-optimizer/assets/23622319/619eef72-84d1-496c-9e0a-24962e5fa4d8

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/gi/ui/src/components/team/TeamCard.tsx [hunk 1]
```diff
@@ -68,17 +73,42 @@ export function TeamCard({
...
+              )}
+            </Typography>
+          </CardActionArea>
+          <TeamDelModal
+            teamId={teamId}
+            show={showDel}
+            onHide={onHideDel}
+            onDel={function (): void {}}
+          />
+          <Button
```
Comment (frzyc) [inline libs/gi/ui/src/components/team/TeamCard.tsx:100]: I would advice using [IconButton](https://mui.com/material-ui/react-button/#icon-button) and keep it as a red icon on the top right.

**Review by frzyc — CHANGES_REQUESTED** [review]

File: libs/gi/ui/src/components/team/TeamCard.tsx (hunk 1 above)
Comment (chengxi600) [reply libs/gi/ui/src/components/team/TeamCard.tsx:100]: Screenshot 2024-05-16 at 11 57 39 PM

something like this look good?

Comment (frzyc) [reply libs/gi/ui/src/components/team/TeamCard.tsx]: Perfect, exactly what I wanted :D

**Review by frzyc — APPROVED** [review]
LGTM

**Pull request merged**